x-pack/plugin/sql/qa/jdbc/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/JdbcTestUtils.java

```diff
@@ -57,4 +84,26 @@ static long convertFromCalendarToUTC(long value, Calendar cal) {
 
         return convertedDateTime.toInstant().toEpochMilli();
     }
+
+    static String asStringTimestampFromNanos(long nanos) {
+        return asStringTimestampFromNanos(nanos, ZoneId.systemDefault());
+    }
+
+    static String asStringTimestampFromNanos(long nanos, ZoneId zoneId) {
+        ZonedDateTime zdt = ZonedDateTime.ofInstant(Instant.ofEpochMilli(toMilliSeconds(nanos)), zoneId);
+        return StringUtils.toString(zdt.withNano((int) (nanos % 1_000_000_000)));
+    }
+
+    static long randomNanos() {
```

Comment:
Wondering if `randomTimeInNanos()` would make sense. When seeing used I found myself wondering if function returns random fractions of a second at nano resolution, or a time expressed in nanos.

Reply:
yep, will be much more clear.